lib/Parser/Parse.cpp

```diff
@@ -11510,11 +11521,52 @@ ParseNodeProg * Parser::Parse(LPCUTF8 pszSrc, size_t offset, size_t length, char
                 this->GetScanner()->Scan();
             }
 
+            if (isMethod && m_token.tk == tkID)
+            {
+                RestorePoint atPid;
+                IdentPtr pidHint = m_token.GetIdentifier(this->GetHashTbl());
+                this->GetScanner()->Capture(&atPid);
+                this->GetScanner()->Scan();
+                if ((pidHint == wellKnownPropertyPids.get || pidHint == wellKnownPropertyPids.set) && NextTokenIsPropertyNameStart())
+                {
+                    // Getter/setter
+                    // Skip the get/set keyword and continue normally
+                    isGetterSetter = true;
+                }
+                else
+                {
+                    // Not a getter/setter; rewind and treat the token as a name.
+                    this->GetScanner()->SeekTo(atPid);
+                }
+            }
+
             // Eat the computed name expression
             if (m_token.tk == tkLBrack && isMethod)
             {
```

Comment:
Is this part of the change solving a new problem that's introduced by changing the extents of the function to include the computed name? It seems like the problems you're describing are not new, so were we just not coming in here?

Reply:
Interesting. I see why matching brackets seems like it should work. I just wonder if it's really bulletproof. Is there a place where we can record the endpoint of the computed name so we can either advance the scanner past it or do a debug-only check that the scanner is arriving at the same place? Maybe a field on the DeferredStub?

Reply:
Seems like it should parse correctly again if we upfront parsed it already. Agree it would be cool to assert the expression.

Reply:
Actually, now that I think of it, I'm worried about this. You can't actually tokenize JS without knowing where expressions begin and end, because that's how we resolve the ambiguous meaning of the forward slash. I wonder if we may really need to use ParseExpr and do what's necessary a. to recognize that the computed name expression belongs to the enclosing context, b. to recall whether the enclosing context is async.

Reply:
Is there enough ambiguity that it would impact finding the next `]`? I had assumed that the tokenizer would handle strings and comments and regex literals, but that otherwise I'd get a sequence of fairly raw tokens, and I'm not aware of any compound tokens that include `[` or `]` that would need to depend on surrounding context.

Reply:
Regex literals are your problem, since those are marked by /. They can’t be tokenized w/o understanding expressions.

Reply:
Ah, so the tokenizer doesn't automatically switch into regex tokenizing mode, like it does for strings? I'll have a play around tomorrow and see if something like `{ [/]/.exec("]")]() {}}` causes issues.

Reply:
Looks like you are right, the above causes a syntax error when defer parsed.

Reply:
I've just added handling for the regexp case, as well as a debug mode sanity check. Note that the sanity check isn't perfect; without the regex handling you could fool it with something like `{ [/]() { return 1}/.toString()]() { return 2;}}` and the defer parse leg will latch onto the string inside the regexp, thinking it to be legitimate.

Reply:
Turns out it won't be considered legitimate, since it expects that once it has parsed the function it should have consumed all of the source, and in the above case it hasn't, so it throws a syntax error. So, if things don't work as we expect that would lead to a bug (runtime syntax error when invoking a correct function) but not otherwise incorrect behavior.

Reply:
So the idea is to decide whether / indicates regex or division based on the last token seen? Does that always work? Can function expressions, lambdas, etc. create confusion? Both syntax coloring and parallel parsing wanted to disambiguate these cases and never managed it to anybody's satisfaction. I'd honestly feel more comfortable having the initial parse record the point where the name ends and just moving the scanner past it.

Reply:
Great points Paul; this logic has gotten very complicated. I'm also a bit worried about the possible bad outcome: a runtime syntax error is certainly better than a security bug, but worse than a controlled crash (because crawler would presumably find the crash for us).

Reply:
The check that I added here was taken from `FastScanFormalsAndBody` which is one of the few places that switches the parser into regexp mode.

Reply:
Ah. That explains why I was suspicious of it: because I probably wrote it.

Reply:
This is throwing several wrenches in. Pretty much agree with Paul, we should really keep track of the computed name expression and skip it that way.